ARROW-5250: [Java] Add javadoc comments to public methods, remove style check suppression.

emkornfield opened the pull request:
Apologies for the large CL.  After this the one last large javadoc fixing of 2019 is to enable checkstyle to types as well and add missing docs to those.

If possible my review preference would be as suggested edits if at all possible but any feedback is welcome.

In addition to javadocs, I also added private constructors to classes containing only static utility methods.

For a small set of test methods I converted them to "private" instead of providing javadocs.

The approach taken was to provide minimal useful javadoc and not necessarily completely formatted with all parameters, etc.  The intent is to enable style checking to "stop" the bleeding so patch authors can provide useful documentation for new API methods going forward.

jacques-n commented:
Nice job Micah. Super appreciated! A few high level comments at this point:

I'm against requirement for javadocs on public methods/classes in tests (I am assuming we have that req based on the changes). Sure, if there is something helpful to comment we should do it but I find many of the comments on tests to be just duplicating the name of the test method for example. In fact, in some cases here the test names have more meaning than the comments.

It also looks like we require a javadoc on methods but don't require a javadoc that includes arguments for that method. It seems like we should require both or neither.

For the public -> private methods, can you submit those changes as a separate patch? They should get more scrutiny and I'm likely to miss some in a patch with this many changes (partly because I find reviews much harder to do in GH than Gerrit... sorry.) Assuming that some of those methods are not test methods, I would also make sure the Dremio team aren't actively relying on those apis.

emkornfield commented:
@jacques-n thanks for the comments.  I pushed an updated version, that 
1.  Splits the changes into 2 commits the first is all the code change (it seems given the small size and trivial nature a separate PR might be overkill but if you still want one I can do it) the second is all the javadoc changes.
2.  Changed the configuration to exclude test directories and reverted all comments in the test directory (I'll admit to some laziness on not putting much effort into test javadocs).
3.  Removed the requirement for main methods to have javadocs (but left the current ones in place).

In regards to:
> It also looks like we require a javadoc on methods but don't require a javadoc that includes arguments for that method. It seems like we should require both or neither.

I disagree:
1.   I think some docs as long as not useless or incorrect are better then documentation that doesn't have parameters.
2.  In many cases documenting params is redundant (doesn't add value).  I'm not claiming that there are some methods that I added that couldn't benefit from parameter documentation, but I tried to be judicious and would prefer to address this on a case-by-case basis.  The [Google Style Guide](https://google.github.io/styleguide/javaguide.html#s7-javadoc) that we use implicitly allows for leaving out parameter documentation and explicit \@return documentation to avoid this redundancy.

If you are amenable, I'd like to resolve the requirement for parameter documentation by:
1.  Getting consensus on the mailing list as to the communities preference for all method javadoc that checkstyle allows.
2.  Following up with a separate PR.  The size of this PR is already large, I apologize for that, and as I stated above, I think it adds value as a stand-alone basis.  Adding the requirement for parameters will only make it larger.

BryanCutler commented on java/flight/src/main/java/org/apache/arrow/flight/FlightStream.java:
```diff
@@ -187,6 +204,11 @@ public void onCompleted() {
   }
 
+  /**
+   * Cancel's sending the stream to a client.
```
no apostrophe, should be "Cancels.."

emkornfield replied:
done.

BryanCutler commented on java/tools/src/main/java/org/apache/arrow/tools/StreamToFile.java:
```diff
@@ -35,6 +35,9 @@
  */
 public class StreamToFile {
+  /**
+   *  Reads a record batchs from in and writes them to out.
```
should be "Reads an Arrow stream from in and writes it to out"

emkornfield replied:
done.

BryanCutler commented on java/vector/src/main/java/org/apache/arrow/vector/VectorSchemaRoot.java:
```diff
@@ -72,6 +92,9 @@ public VectorSchemaRoot(Schema schema, List<FieldVector> fieldVectors, int rowCo
   }
 
+  /**
+   * Creates a new set of vectors corresponding to schema that are empty.
```
maybe this would sound better "Creates a new set of empty vectors corresponding to the given schema"?

emkornfield replied:
Seems reasonable. changed (will be in the next push as with addressing your other comments.).

BryanCutler commented on java/vector/src/main/java/org/apache/arrow/vector/VectorSchemaRoot.java:
```diff
@@ -161,6 +185,9 @@ private void printRow(StringBuilder sb, List<Object> row) {
   }
 
+  /**
+   * Returns a a tab separated value of vectors (based on their java object representation).
```
typo: "a a"

emkornfield replied:
fixed.

BryanCutler commented on java/vector/src/main/java/org/apache/arrow/vector/VectorUnloader.java:
```diff
@@ -26,22 +26,39 @@
 import io.netty.buffer.ArrowBuf;
 
+/**
+ * Helper class the handles converting a vector the a {@link ArrowRecordBatch}.
```
typo: the -> that

emkornfield replied:
done.

BryanCutler commented on java/vector/src/main/java/org/apache/arrow/vector/ipc/ArrowWriter.java:
```diff
@@ -98,6 +98,9 @@ public void start() throws IOException {
   }
 
+  /**
+   * Write the vector this vector was constructed with.
```
Maybe "Write the record batch currently loaded in the VectorSchemaRoot"?

emkornfield replied:
Revised to  "Writes the record batch currently loaded in this instance's VectorSchemaRoot."

BryanCutler commented on java/vector/src/main/java/org/apache/arrow/vector/ipc/JsonFileReader.java:
```diff
@@ -69,13 +69,23 @@
+/**
+ * A reader for integration test JSON files that converts translates them into Vectors.
+ * This class uses a streaming parser API, method naming tends to reflect this implementation
+ * detail.
```
How about "A reader for JSON files that translates them into vectors. This reader is used for integration tests...."

emkornfield replied:
SGTM.

emkornfield commented:
@BryanCutler thanks for the review, addressed your comments.

@jacques-n Please let me know if I have addressed yours sufficiently.

BryanCutler commented on java/flight/src/main/java/org/apache/arrow/flight/FlightClient.java:
```diff
@@ -99,7 +102,7 @@ public FlightClient(BufferAllocator incomingAllocator, Location location) {
 
   /**
-   * List actions available on the Flight service.
+   * Lists actions available on the Flight service.
```
I don't have the best grammar, but I thought it was fine before.. as in "Calling this method will list actions available..."  maybe it could go either way?

emkornfield replied:
Neither do I.  In this case the change is dictated by my interpretation of the style guide: https://google.github.io/styleguide/javaguide.html#s7.2-summary-fragment

BryanCutler replied:
Ok seems fine then

BryanCutler commented on java/flight/src/main/java/org/apache/arrow/flight/FlightClient.java:
```diff
@@ -112,7 +115,7 @@ public FlightClient(BufferAllocator incomingAllocator, Location location) {
 
   /**
-   * Perform an action on the Flight service.
+   * Performs an action on the Flight service.
```
same

emkornfield replied:
See above.  style guide.

BryanCutler commented on java/flight/src/main/java/org/apache/arrow/flight/FlightClient.java:
```diff
@@ -123,13 +126,16 @@ public FlightClient(BufferAllocator incomingAllocator, Location location) {
 
   /**
-   * Authenticate against the Flight service.
+   * Authenticates against the Flight service.
```
same

emkornfield replied:
See above.

BryanCutler commented on java/flight/src/main/java/org/apache/arrow/flight/FlightClient.java:
```diff
@@ -123,13 +126,16 @@ public FlightClient(BufferAllocator incomingAllocator, Location location) {
   }
 
+  /**
+   * Authenticates with a username an password.
```
an -> and

Also maybe Authenticates -> Authenticate

emkornfield replied:
fixed in pending commit.

BryanCutler commented on java/flight/src/main/java/org/apache/arrow/flight/FlightInfo.java:
```diff
@@ -35,13 +35,24 @@
   private final long records;
 
+  /**
+   * Constructs a new instance
```
add a new line here to be consistent. Can there be a checkstyle rule for that?

emkornfield replied:
fixed in pending commit.

BryanCutler commented on java/flight/src/main/java/org/apache/arrow/flight/example/ExampleTicket.java:
```diff
@@ -41,6 +44,12 @@
   private final String uuid;
 
+  /**
+   * Constructs a new instance
```
newline

emkornfield replied:
done.

BryanCutler commented on java/flight/src/main/java/org/apache/arrow/flight/example/InMemoryStore.java:
```diff
@@ -60,6 +66,9 @@ public void getStream(CallContext context, Ticket ticket,
   }
 
+  /**
+   * Returns the appropriate stream give the ticket (streams are indexed by path and an ordinal).
```
give -> given

emkornfield replied:
done.

BryanCutler commented on java/vector/src/main/java/org/apache/arrow/vector/TypeLayout.java:
```diff
@@ -217,11 +220,16 @@ public TypeLayout(BufferLayout... bufferLayouts) {
   }
 
+  /**
+   * Returns the types of each buffer for in this layout.
```
remove for or in?

emkornfield replied:
removed in.  added a clarifying sentence as a follow-up.

BryanCutler commented on java/vector/src/main/java/org/apache/arrow/vector/UInt4Vector.java:
```diff
@@ -228,6 +240,8 @@ public void set(int index, int isSet, int value) {
     }
   }
 
+  /** Same as {@link #set(int, int, int)} but will reallocate if the buffer if index
```
I think multi-line comment should be
```
/**
 * Same as
 * is larger..
 */

BryanCutler commented on java/vector/src/main/java/org/apache/arrow/vector/complex/ListVector.java:
```diff
@@ -523,6 +544,8 @@ public UnionListReader getReader() {
     return reader;
   }
 
+
```
remove extra newline I think

emkornfield replied:
done.

jacques-n commented:
Hey @emkornfield, I feel like you've addressed my comments. Thanks, I appreciate it.

emkornfield commented:
@BryanCutler thanks for the detailed review.  I pushed a new commit addressing your feedback.

emkornfield commented:
@BryanCutler if this look good to you, would you mind if I merged this?  I'm afraid there might be race conditions with other in flight PRs that might not have all the necessary javadocs but get merged anyways.
